src/dev/build/tasks/clean_tasks.js

```diff
@@ -57,12 +57,90 @@ export const CleanExtraFilesFromModulesTask = {
   description: 'Cleaning tests, examples, docs, etc. from node_modules',
 
   async run(config, log, build) {
+
     await deleteAll(log, [
-      build.resolvePath('node_modules/**/test/**/*'),
-      build.resolvePath('node_modules/**/tests/**/*'),
-      build.resolvePath('node_modules/**/example/**/*'),
-      build.resolvePath('node_modules/**/examples/**/*'),
-    ]);
+      'test/**/*',
+      'tests/**/*',
+      '__tests__/**/*',
+      'mocha.opts',
+      '*.test.js',
+      '*.snap',
+      'coverage/**/*',
+
+      'doc/**/*',
+      'docs/**/*',
+
+      'example/**/*',
+      'examples/**/*',
+      'demo/**/*',
+      'samples/**/*',
+
+      '.bin/**/*',
+
+      '*.map',
+      '*.coffee',
+      '*.flow',
+      '*.webidl',
+      '*.exe',
+      '*.bat',
+      '*.zip',
+      '*.gz',
+      '*.sh',
+      '*.scss',
+      '*.sass',
+      '*.png',
```

Comment:
Note that x-pack is now in node_modules on builds, so I don't think removing things like images is safe.